release/docker/Dockerfile

```diff
@@ -6,10 +6,12 @@ RUN useradd -mU mitmproxy
 RUN apt-get update \
     && apt-get install -y --no-install-recommends gosu \
     && rm -rf /var/lib/apt/lists/*
-    
+
 COPY $MITMPROXY_WHEEL /home/mitmproxy/
-RUN pip3 install --no-cache-dir -U /home/mitmproxy/${MITMPROXY_WHEEL} \
-    && rm -rf /home/mitmproxy/${MITMPROXY_WHEEL}
+COPY requirements.txt /home/mitmproxy/
+RUN pip3 install --no-cache-dir --require-hashes -r /home/mitmproxy/requirements.txt \
+    && pip3 install --no-cache-dir --no-deps /home/mitmproxy/${MITMPROXY_WHEEL} \
+    && rm /home/mitmproxy/${MITMPROXY_WHEEL} /home/mitmproxy/requirements.txt
```

Comment:
Our Docker images now also only install pinned dependencies.

Reply:
Should we pin the container itself as well?